presto-main/src/main/java/com/facebook/presto/metadata/Metadata.java

```diff
@@ -114,6 +116,29 @@
     Map<QualifiedObjectName, List<ColumnMetadata>> listTableColumns(Session session, QualifiedTablePrefix prefix);
 
     /**
+     * Gets the TableIdentity for the specified table.
+     */
+    @NotNull
+    TableIdentity getTableIdentity(Session session, TableHandle tableHandle);
+
+    /**
+     * Gets the ColumnIdentity for all columns of the specified table.
+     */
+    @NotNull
+    Map<String, ColumnIdentity> getColumnIdentities(Session session, TableHandle tableHandle);
+
+    /**
+     * Deserialize the bytes to TableIdentity
+     */
+    TableIdentity deserializeTableIdentity(Session session, String catalogName, byte[] bytes);
+
+    /**
+     * Deserialize the bytes to ColumnIdentity
+     */
+    @NotNull
+    Map<String, ColumnIdentity> deserializeColumnIdentities(Session session, String catalogName, Map<String, byte[]> serializedColumnIdentities);
```

Comment:
Why not just have `ColumnIdentity deserializeColumnIdentitie(Session session, String catalogName, byte[] bytes)`